packages/data/src/world/mq/dodongo_cavern_mq.yml

```diff
@@ -2,159 +2,279 @@
-    "MQ Dodongo Cavern Pot Miniboss 2": "true"
-    "MQ Dodongo Cavern Pot Miniboss 3": "true"
-    "MQ Dodongo Cavern Pot Miniboss 4": "true"
-"Dodongo Cavern Upper Ledges":
+    "MQ Dodongo Cavern Heart Vanilla Bomb Bag Room": "true"
+    # Fails to load
+    "MQ Dodongo Cavern Grass Vanilla Bomb Bag Room": "can_cut_grass"
+"Dodongo Cavern Vanilla Bomb Bag Room Upper":
   dungeon: DC
   exits:
+    "Dodongo Cavern Vanilla Bomb Bag Room Lower": "true"
+    "Dodongo Cavern Main Room Upper Skull Side": "event(DC_MQ_SHORTCUT) || climb_anywhere"
     "Dodongo Cavern Room After Upper Lizalfos": "true"
-    "Dodongo Cavern Torch Room": "true"
   events:
-    "DC_MQ_SHORTCUT": "true"
+    DC_MQ_SHORTCUT: "has_bombflowers || has_fire || has_explosives_or_hammer"
   locations:
     "MQ Dodongo Cavern Upper Ledge Chest": "true"
     "MQ Dodongo Cavern Pot Vanilla Bomb Bag Room 1": "true"
     "MQ Dodongo Cavern Pot Vanilla Bomb Bag Room 2": "true"
+"Dodongo Cavern MQ Larvae Room":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Vanilla Bomb Bag Room Lower": "event(MQ_DC_CLEAR_LARVAE_ROOM)"
+  events:
+    MQ_DC_CLEAR_LARVAE_ROOM: "soul_enemy(SOUL_ENEMY_GOHMA_LARVA) && can_damage"
+  locations:
+    "MQ Dodongo Cavern Larvae Room Chest": "event(MQ_DC_CLEAR_LARVAE_ROOM)"
+    "MQ Dodongo Cavern GS Larve Room": "gs && can_damage_skull"
+"Dodongo Cavern Room Before Upper Lizalfos":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Upper Lizalfos": "(can_use_sticks && has_goron_bracelet) || has_explosives_or_hammer || has_blue_fire || has_blue_fire_arrows_mudwall" #Needs macro for getting past enemies still in path
+    "Dodongo Cavern Vanilla Bomb Bag Room Lower": "has_fire"
+  locations:
+    "MQ Dodongo Cavern Pot Before Miniboss 1": "true"
+    "MQ Dodongo Cavern Pot Before Miniboss 2": "true"
     # Fails to load
-    "MQ Dodongo Cavern Grass Vanilla Bomb Bag Room": "can_cut_grass"
+    "MQ Dodongo Cavern Grass Room Before Miniboss": "can_cut_grass"  
+"Dodongo Cavern Room After Upper Lizalfos":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Upper Lizalfos": "true"
+    "Dodongo Cavern Vanilla Bomb Bag Room Upper": "(has_goron_bracelet && (has_weapon || can_use_sticks || has_ranged_weapon)) || has_explosives_or_hammer || is_adult || climb_anywhere || hookshot_anywhere"
+  locations:
+    "MQ Dodongo Cavern Pot After Miniboss 1": "true"
+    "MQ Dodongo Cavern Pot After Miniboss 2": "true"
+"Dodongo Cavern Upper Lizalfos":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Room Before Upper Lizalfos": "event(MQ_DC_UPPER_LIZALFOS_CLEAR)"
+    "Dodongo Cavern Room After Upper Lizalfos": "event(MQ_DC_UPPER_LIZALFOS_CLEAR)"
+    #"Dodongo Cavern Lower Lizalfos": "has_explosives_or_hammer || hookshot_anywhere || climb_anywhere"
+  events:
+    MQ_DC_UPPER_LIZALFOS_CLEAR: "soul_lizalfos_dinalfos && (can_use_sticks || has_weapon || can_use_slingshot)"
+  locations:
+    "MQ Dodongo Cavern GS Upper Lizalfos": "gs && (has_explosives_or_hammer || longshot_anywhere)"
```

Comment:
Should probably include the climb anywhere method for this, assuming you can climb up the pillar after climbing over the rocks.

Reply:
You cannot do that.

Reply:
Yes you can. Tested it before i did the time travel.

Reply:
Then I don't know what you mean. What line are you even referring to?

Reply:
This one right here. GS Upper Lizalfos. You can climb around the rocks, drop down, then climb up the pillar.

Reply:
But that's going from Upper>Lower>Upper GS, not Upper>GS; thats why the Upper GS is listed in the lower region too

Reply:
But there is no exit between the regions here due to the lizalfos fights. So need to account for it here as well. The way i did it in vanilla was putting the time travel requirement on the exits (because there are not yet any fight events). Here, you could require the lower fight events to go the lower region, and vice versa. But you would still need this climb anywhere method in the upper region cause you may not have song of time.

Reply:
That's what I did, though? And what does that have to do with the GS?

Reply:
No, that has not been done yet. There are currently no exits going between the lower and upper lizalfos fights. And it has to do with the GS because you may not have SoT to go between the two regions logically.

Reply:
I do have the exits there, just commented out because of the lizalfos thing. now that the time travel is in, could they be implemented and then it would work just fine? Lines 149 and 189

Reply:
They can be implemented, but they would need to require the relevant events to go through the doors in their respective region. Cause otherwise you could exit to them and be expected to go through the doors when you dont have Song of Time due to having access to the event that does not require SoT. And because they would need to require the events, since you may not be able to do the events from where you entered, you need to be able to account for that with the locations in these regions (already done for everything but climb anywhere for this GS from upper).

Reply:
Okay added climb_anywhere to the GS from Upper